server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -715,23 +715,29 @@ private synchronized void cleanCache(double stalenessThreshold) {
             }
             // Contains CleanupKey objects with open shard but invalidated readerCacheKeyId.
             final Set<CleanupKey> cleanupKeysFromOutdatedReaders = new HashSet<>();
-            // Contains CleanupKey objects of a closed shard.
+            // Contains CleanupKey objects for a full cache cleanup.
+            final Set<Tuple<ShardId, Integer>> cleanupKeysFromFullClean = new HashSet<>();
```

Comment:
I'm not entirely sure why we can't do that, this is the way it was done by @kiranprakash154 and @sgup432 for the original cache cleanup logic. They had just combined cleanup keys for a full cleanup or for a single closed shard into `cleanupKeysFromClosedShards`. 

Here we're just separating them out because we only want to drop the dimensions from the stats object if the shard is closed, not if the whole cache is cleaned.